exasol_integration_test_docker_environment/lib/test_environment/database_waiters/is_database_ready_thread.py

```diff
@@ -28,37 +32,46 @@ def __init__(self,
         self.output_db_connection = None
         self.output_bucketfs_connection = None
         self.docker_db_image_version = docker_db_image_version
+        self.executor_factory = executor_factory
 
     def stop(self):
         self.logger.info("Stop IsDatabaseReadyThread")
         self.finish = True
 
     def run(self):
-        db_connection_command = ""
-        bucket_fs_connection_command = ""
+        print(self.__class__, "run", dir(self.executor_factory))
         try:
-            exaplus_path = find_exaplus(self._db_container)
-            db_connection_command = self.create_db_connection_command(exaplus_path)
-            bucket_fs_connection_command = self.create_bucketfs_connection_command()
-        except RuntimeError as e:
-            self.logger.error(e)
+            with self.executor_factory.executor() as executor:
+                print(self.__class__, "executor")
+                db_connection_command = ""
```

Comment:
Also a candidate for extraction of function for better readability. Probably a separate PR